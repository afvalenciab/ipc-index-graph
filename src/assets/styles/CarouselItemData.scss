.carousel--item_data {
  display: flex;
  justify-content: space-between;
  width: 200px;
  height: 70px;
  border-radius: 10px;
  box-sizing: border-box;
  padding: 10px;
  margin-right: 8px;
  flex-shrink: 0;
  background: linear-gradient(90deg,#2b2b2b, #171717);

  h2 {
    margin: 0;
    font-size: 16px;
    color: #FFF;
  }

  .item--data {
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    p {
      margin: 0;
    }

    .price {
      color: #FFF;
      font-weight: 600;
    }

    .percentage {
      text-align: end;
    }
    .percentage.up {
      color: #00E464;
    }
    .percentage.down {
      color: #FD5F5D;
    }
  }
}