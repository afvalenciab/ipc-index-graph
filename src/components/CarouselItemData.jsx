import React from 'react';
import '../assets/styles/CarouselItemData.scss';

const CarouselItemData = (props) => {
  const { nameCard, ipcValue } = props;
  const symbolPercentage = ipcValue ? ipcValue.Porcentaje >= 0 : false;

  return (
    <div className='carousel--item_data'>
      <h2>{nameCard}</h2>
      <div className='item--data'>
        <p className='price'>{ipcValue ? new Intl.NumberFormat('de-DE').format(ipcValue.Precio) : '-'}</p>
        {ipcValue && (
          <p className={`percentage ${symbolPercentage ? 'up' : 'down'}`}>
            {`${ipcValue.Porcentaje >= 0 ? '+' : ''}${ipcValue.Porcentaje}%`}
          </p>
        )}
      </div>
    </div>
  );
};

export default CarouselItemData;
