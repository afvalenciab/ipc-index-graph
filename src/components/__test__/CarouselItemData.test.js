import React from 'react';
import { create } from 'react-test-renderer';
import CarouselItemData from '../CarouselItemData';
import ProviderMock from '../../__mocks__/ProviderMocks';

describe('CarouselItemData testing', () => {
  test('Match Snapshot', () => {
    const carouselItemData = create(
      <ProviderMock>
        <CarouselItemData />
      </ProviderMock>
    );
    expect(carouselItemData.toJSON()).toMatchSnapshot();
  });
});