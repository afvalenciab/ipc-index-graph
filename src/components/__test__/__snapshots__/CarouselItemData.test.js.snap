// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselItemData testing Match Snapshot 1`] = `
<div
  className="carousel--item_data"
>
  <h2 />
  <div
    className="item--data"
  >
    <p
      className="price"
    >
      -
    </p>
  </div>
</div>
`;
